src/python/pants/jvm/subsystems.py

```diff
@@ -74,6 +74,10 @@ def default_resolve(self) -> str:
     def debug_args(self) -> tuple[str, ...]:
         return cast("tuple[str, ...]", tuple(self.options.debug_args))
 
+    def is_jvm_target(self, target: Target) -> bool:
```

Comment:
Nit: maybe `has_resolves`? Or alternatively, making a fallible version of `def resolves_for_target`, which returns `None` rather than raising the assertion?
```python
def resolves_for_target(self, target: Target) -> tuple[str, ...]:
    resolves = self.resolves_for_target_optional(target)
    if resolves is None:
        raise AssertionError(..)

def resolves_for_target_optional(self, target: Target) -> tuple[str, ...] | None:
    ...
```